streaming/src/main/scala/org/apache/spark/streaming/util/BatchedWriteAheadLog.scala

```diff
@@ -122,6 +122,7 @@ private[util] class BatchedWriteAheadLog(val wrappedLog: WriteAheadLog, conf: Sp
   override def close(): Unit = {
     logInfo(s"BatchedWriteAheadLog shutting down at time: ${System.currentTimeMillis()}.")
     synchronized {
+      if (!active) return
```

Comment:
`active` might be simpler as an `AtomicBoolean`. Then I think you can just write `if (!active.getAndSet(false)) return` as the very first line.